### PR TITLE
[HOPSWORKS-1257] sudoers script should be owned by root and 700

### PR DESCRIPTION
**Pull request by SirOibaf**

### CODE REVIEW

File: providers/sudoers.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        group 'root'
+        mode '0711'
+        action :create
+    end
+
+    template "#{node["install"]["sudoers"]["scripts_dir"]}/#{new_resource.script_name}" do
+        source new_resource.template
+        owner  'root'
+        group  new_resource.group 
+        mode   '071'
```
Comment (kouzant) [inline providers/sudoers.rb]: I guess this should be `0711`

File: templates/default/config.ini.erb [hunk 2]
```diff
@@ -25,14 +26,16 @@ hostname = <%= @hostname %>
 group-name = <%= node[:kagent][:group_name] %>
 hadoop-home = <%= @hops_dir %>/hadoop
 certs-dir = <%= node[:kagent][:certs_dir] %>
+certs-user = <%= node['kagent']['certs_user'] %>
```
Comment (kouzant) [inline templates/default/config.ini.erb]: Can you also fix this test https://github.com/logicalclocks/kagent-chef/blob/master/files/default/kagent_utils/tests/test_kagent_config.py and run `pytest` ?

File: attributes/default.rb [hunk 3]
```diff
@@ -4,7 +4,9 @@
 default["kagent"]["version"]                       = node["install"]["version"]
 default["kagent"]["user"]                          = node["install"]["user"].empty? ? "kagent" : node["install"]["user"]
 default["kagent"]["group"]                         = node["install"]["user"].empty? ? "kagent" : node["install"]["user"]
+
 default["kagent"]["certs_group"]                   = "certs"
+default["kagent"]["certs_user"]                    = "certs"
```
Comment (kouzant) [inline attributes/default.rb]: Add it to metadata for completeness

File: recipes/default.rb [hunk 4]
```diff
@@ -246,21 +250,11 @@
   code <<-EOH                                                                                                       
        openssl rsa -in #{node['kagent']['certs_dir']}/priv.key -out #{node['kagent']['certs_dir']}/priv.key.rsa
        chmod 640 #{node['kagent']['certs_dir']}/priv.key.rsa
-       chown root:#{node['kagent'['certs_group']]} #{node['kagent']['certs_dir']}/priv.key.rsa
+       chown #{node['kagent']}:#{node['kagent'['certs_group']]} #{node['kagent']['certs_dir']}/priv.key.rsa
```
Comment (kouzant) [inline recipes/default.rb]: The user is wrong and in the group there is a missing right bracket

File: templates/default/config.ini.erb [hunk 5]
```diff
@@ -25,14 +26,16 @@ hostname = <%= @hostname %>
 group-name = <%= node[:kagent][:group_name] %>
 hadoop-home = <%= @hops_dir %>/hadoop
```
[the remark this answers is not in the document]
Comment (SirOibaf) [reply templates/default/config.ini.erb]: Yes, removed also from Hopsworks.

File: providers/sudoers.rb (hunk 1 above)
Comment (SirOibaf) [reply providers/sudoers.rb]: Actually I wanted to put 710.

File: recipes/default.rb (hunk 4 above)
Comment (SirOibaf) [reply recipes/default.rb]: thanks

File: files/default/agent.py [hunk 6]
```diff
@@ -371,15 +371,14 @@ def _log_conda_command(self, project_name, operation, artifact, artifact_version
...
         command_id = command['id']
         op = command['op'].upper()
         proj = command['proj']
         install_jupyter = str(command['installJupyter']).lower()
 
         tempfile_fd = None
-        script = kconfig.bin_dir + "/anaconda_env.sh"
-        logger.info("sudo {0} {1} {2} {3} {4} '{5}' {6} {7}".format(script, user, op, proj, arg, offline, kconfig.hadoop_home, install_jupyter))
+        script = kconfig.sbin_dir + "/anaconda_env.sh"
+        logger.info("sudo -u {1} {0} {2} {3} {4} '{5}' {6} {7}".format(script, kconfig.conda_user, p, proj, arg, offline, kconfig.hadoop_home, install_jupyter))
```
[the remark this answers is not in the document]
Comment (SirOibaf) [reply files/default/agent.py]: The lazyness is powerful in me.

File: attributes/default.rb (hunk 3 above)
Comment (SirOibaf) [reply attributes/default.rb]: done.

**Pull request merged**